### PR TITLE
feat(http): add swagger definitions for influxql query

### PR DESCRIPTION
**Pull request by goller**
This is documentation for the 1.0 compatibility query path.

### CODE REVIEW

File: http/swagger.yml [hunk 1]
```diff
@@ -5,6 +5,138 @@ info:
 servers:
   - url: /v1
 paths:
+  /query:
+   servers:
```
Comment (jademcgough) [inline http/swagger.yml]: Is this supposed to be here? It's not an HTTP verb.

File: http/swagger.yml [hunk 2]
```diff
@@ -5,6 +5,138 @@ info:
...
+            type: string
+          description: optional retention policy to query (> 1.6.0)
+        - in: query
+          name: bucket
+          schema:
+            type: string
+          description: bucket is the query and write destination store
+        - in: query
+          name: precision
+          description: preciscion rounds the timestamp to the nearest unit
```
Comment (jademcgough) [inline http/swagger.yml]: precision

File: http/swagger.yml [hunk 3]
```diff
@@ -5,6 +5,138 @@ info:
...
+          schema:
+            type: string
+            description: acceptable epochs define the time units returned in the query
```
Comment (jademcgough) [inline http/swagger.yml]: looks like a copy-paste of the descriptions from the epoch schema

File: http/swagger.yml [hunk 4]
```diff
@@ -5,6 +5,138 @@ info:
...
+              "m",
+              "h"
+            ]
+        - in: query
+          name: epoch
+          description: override default RFC3339 time format with unix epoch at a precision; epoch rounds the timestamp to the nearest unit
+          schema:
+            type: string
+            description: acceptable epochs define the time units returned in the query
+            enum: [
```
Comment (jademcgough) [inline http/swagger.yml]: Why make this be an enum? I'd prefer boolean, and use `precision` to determine the precision.

Comment (jademcgough) [reply http/swagger.yml]: Alternatively, we could have a `format` param if we expect to support more than unix epoch and rfc3339 (though this seems unnecessary for now).

File: http/swagger.yml [hunk 5]
```diff
@@ -5,6 +5,138 @@ info:
...
+            enum:
+             - gzip
+        - in: header
+          name: X-Request-Id
+          description: specifies correlating requests across services
+          schema:
+            type: string
+            format: uuid
+    responses:
+      '200':
```
Comment (nhaugo) [inline http/swagger.yml]: 204 is used in, at least, the ping endpoint

Comment (nhaugo) [reply http/swagger.yml]: And writes

File: http/swagger.yml [hunk 6]
```diff
@@ -774,3 +906,72 @@ components:
           readOnly: true
           type: string
       required: [code, message]
+    InfluxQLResults:
+      properties:
+        error:
```
Comment (jademcgough) [inline http/swagger.yml]: fyi we have an `Error` component that could be used here (just mentioning, I have no preference other than eventual consistency in our API).

File: http/swagger.yml (hunk 1 above)
Comment (goller) [reply http/swagger.yml]: @jademcgough yup, it overrides /v1 to / so, in theory the path is `/query`

File: http/swagger.yml (hunk 3 above)
Comment (goller) [reply http/swagger.yml]: @jademcgough , yeah, as far as I can tell from the code, one could use either parameter for some rason.

File: http/swagger.yml (hunk 1 above)
Comment (jademcgough) [reply http/swagger.yml]: 👍

File: http/swagger.yml (hunk 6 above)
Comment (goller) [reply http/swagger.yml]: @jademcgough good point.  What do you think of me putting a `oneOf` in the error response schema on the endpoint for this error and our new style error?  

I put it here just for backwards compat.

File: http/swagger.yml (hunk 3 above)
Comment (jademcgough) [reply http/swagger.yml]: That's fine, I just mean that the description should be `acceptable precisions` instead of `acceptable epochs`.

**Comment by jademcgough** [comment]
Feel free to ignore most of my comments, I hadn't realized that this is documentation of existing code.

File: http/swagger.yml (hunk 3 above)
Comment (goller) [reply http/swagger.yml]: Fixed in ae1b03e

**Comment by goller** [comment]
@jademcgough I'll do a follow-on PR for the new-school query

File: http/swagger.yml (hunk 6 above)
Comment (jademcgough) [reply http/swagger.yml]: don't worry about this, then :3

**Pull request merged**